airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java

```diff
@@ -178,39 +185,58 @@ private void checkImport(Path tempFolder) throws IOException, JsonValidationExce
     }
   }
 
-  private <T> void importConfigsFromArchive(final Path sourceRoot, final boolean dryRun) throws IOException, JsonValidationException {
-    List<String> sourceDefinitionsToMigrate = new ArrayList<>();
-    List<String> destinationDefinitionsToMigrate = new ArrayList<>();
+  private <T> void importConfigsFromArchive(final Path sourceRoot, Optional<Path> seedPath, final boolean dryRun)
+      throws IOException, JsonValidationException {
+    final List<String> sourceDefinitionsToMigrate = new ArrayList<>();
+    final List<String> destinationDefinitionsToMigrate = new ArrayList<>();
     final boolean[] sourceProcessed = {false};
     final boolean[] destinationProcessed = {false};
-    List<String> directories = listDirectories(sourceRoot);
+    final List<String> directories = listDirectories(sourceRoot);
     // We sort the directories cause we want to process SOURCE_CONNECTION before
     // STANDARD_SOURCE_DEFINITION and DESTINATION_CONNECTION before STANDARD_DESTINATION_DEFINITION
     // so that we can identify which definitions should not be upgraded to the latest version
     Collections.sort(directories);
-    Map<ConfigSchema, Stream<T>> data = new LinkedHashMap<>();
-    Map<ConfigSchema, Map<String, T>> latestSeeds = latestSeeds();
+    final Map<ConfigSchema, Stream<T>> data = new LinkedHashMap<>();
+
+    final Map<ConfigSchema, Map<String, T>> seed;
+    if (seedPath.isPresent()) {
```

Comment:
NIT: `seedPath.map(getSeed).orElse(new HashMap)`

Reply:
that doesn't work because getSeed throws an `IOException`.